### ISSUE TITLE
Requesting `log_softmax`, `log_sigmod`, `log_sum_exp`

### ISSUE THREAD

**Issue by boathit** [FluxML/NNlib.jl#21]
The output of `softmax` will be `0` when the number is very close to `0` especially in the case that it is represented by `Float32` in `CuArray`, and it will then leads to an infinity loss if we use the output to compute the cross entropy loss. It will be more numerically stable if `NNlib` could provide `log_softmax` function.

**Comment by MikeInnes** [comment]
It should be easy enough to write get these working, any chance you can take a crack at it?

**Comment by tejank10** [comment]
Hey, I would like to take up this issue.

**Comment by tejank10** [comment]
Hi @MikeInnes ! I have submitted PR [#152](https://github.com/FluxML/Flux.jl/pull/152) in Flux to support gradients on the flux side. Could you please review it? Thanks.
